server/datastore/s3/carves.go

```diff
@@ -0,0 +1,236 @@
+package s3
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/fleetdm/fleet/server/kolide"
+	"github.com/pkg/errors"
+)
+
+const (
+	defaultMaxS3Keys = 1000
+	cleanupSize      = 1000
+	// This is Golang's way of formatting timestrings, it's confusing, I know.
+	// If you are used to more conventional timestrings, this is equivalent
+	// to %Y/%m/%d/%H (year/month/day/hour)
+	timePrefixFormat = "2006/01/02/15"
```

Comment:
Yes, because S3 will always return list of objects sorted by their key (and no way to change that at the time of writing). So this allows me to effectively list files in creation order up to hour precision. It's useful in the cleanup method.